pkg/registry/apis/dashboard/legacy_storage.go

```diff
@@ -10,16 +10,20 @@ import (
 	"k8s.io/apiserver/pkg/registry/generic/registry"
 	"k8s.io/apiserver/pkg/registry/rest"
 
+	"github.com/grafana/grafana/pkg/apimachinery/identity"
 	"github.com/grafana/grafana/pkg/apimachinery/utils"
 	grafanaregistry "github.com/grafana/grafana/pkg/apiserver/registry/generic"
 	grafanarest "github.com/grafana/grafana/pkg/apiserver/rest"
 	"github.com/grafana/grafana/pkg/registry/apis/dashboard/legacy"
+	"github.com/grafana/grafana/pkg/services/dashboards"
 	"github.com/grafana/grafana/pkg/storage/unified/apistore"
 	"github.com/grafana/grafana/pkg/storage/unified/resource"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
 type DashboardStorage struct {
-	Access legacy.DashboardAccess
+	Access           legacy.DashboardAccess
+	DashboardService *dashboards.DashboardService
```

Comment:
Since this is an interface, you don't need a pointer to it:
```suggestion
	DashboardService dashboards.DashboardService
```